### DISCUSSION TITLE
About the POTCAR versions used in keys of `functional_dir` dictionary without version numbers.

### DISCUSSION THREAD

**Discussion by hongyi-zhao** [materialsproject/pymatgen#3904] [General]
As shown below, there are some keys used in `functional_dir` dict which don't include version numbers in their names, aka, `PBE`, `LDA` and `LDA_US`:
https://github.com/materialsproject/pymatgen/blob/d77f77697c7298fd0596e3aa55689709445bdf0e/src/pymatgen/io/vasp/inputs.py#L1757-L1774

So, I would like to know, when using these keys to set the `user_potcar_functional` or `potcar_functional`, what exact versions of the corresponding POTCARs should be used?

Regards,
Zhao

**Comment by DanielYang59** [comment]
Hi,

I think this is very similar to https://github.com/materialsproject/pymatgen/issues/3951. `functional_dir` provides a mapping from the functional you pick (dict key) to the local directory (dict value) where `pymatgen` would try to search for POTCARs: https://github.com/materialsproject/pymatgen/blob/b28c9371d684b8a605830cd13060bef9accf68cf/src/pymatgen/io/vasp/inputs.py#L2278-L2279

I.e. you get whatever you put at the nominated location.

**Reply by hongyi-zhao** [reply]
However, in reality, we all name these directories based on specific versions, so is it meaningful to do so?

**Reply by DanielYang59** [reply]
Emm... Looks like that's one of the most ancient piece of code of `pymatgen` (added in b2948d729f074666ea61681a14958bbc0e0060a8 2011), and my guess is (I could be wrong here):
- `POT_GGA_PAW_PBE`/`POT_LDA_PAW` are the default VASP PSP directory name (I failed to find a PSP directory naming convention from VASP wiki; also I'm not sure if that's the case for all versions of VASP), and it would therefore be a good default location to search
- VASP 5.4.4 was not released back that time (released in 2017), and backward compatibility is certainly playing a part here (we cannot change the default dir from `POT_GGA_PAW_PBE` to `POT_GGA_PAW_PBE_XX` after a new VASP release which would cause breakage)
